app/src/main/java/io/reark/rxgithubapp/advanced/data/DataLayer.java

```diff
@@ -38,13 +38,14 @@
 import io.reark.rxgithubapp.shared.network.GitHubService;
 
 import static io.reark.reark.utils.Preconditions.checkNotNull;
-import static io.reark.reark.utils.Preconditions.get;
 
 public class DataLayer extends ClientDataLayerBase {
     private static final String TAG = DataLayer.class.getSimpleName();
 
     private final Context context;
 
+    private int nextListenerIndex = 0;
```

Comment:
I think a random GUID would be better: If I'm not mistaken this is the client part of the DataLayer and it can be run from may processes. The ids assigned in different processes would then collide.